### PR TITLE
Add support for getting the current thread name and id

### PR DESCRIPTION
**Pull request by 0xZOne**
Add support for getting the current thread name and the system-wide id.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: fml/thread.cc [hunk 1]
```diff
@@ -67,6 +73,14 @@ void Thread::SetCurrentThreadName(const Thread::ThreadConfig& config) {
   SetThreadName(config.name);
 }
 
+std::string Thread::GetCurrentName() {
+  return *(tls_thread_name.get());
+}
+
+ThreadId Thread::GetCurrentId() {
+  return std::this_thread::get_id();
```
Comment (0xZOne) [inline fml/thread.cc:81]: Here maybe we should call something like `syscall(__NR_gettid)` or `pthread_threadid_np` to get the system-wide unique integral ID of the thread.

File: fml/thread.cc [hunk 2]
```diff
@@ -67,6 +77,32 @@ void Thread::SetCurrentThreadName(const Thread::ThreadConfig& config) {
   SetThreadName(config.name);
 }
 
+std::string Thread::GetCurrentName() {
+  return *(tls_thread_name.get());
```
Comment (0xZOne) [inline fml/thread.cc:81]: We may also need to consider the threads created by the platform, like Android.

**Comment by chinmaygarde** [comment]
I'd like to understand why this is necessary. We don't use FML for managing all threads in the process. So the use of the TLS slot becomes problematic. Can you add a link to the issue you are trying to fix?

**Comment by 0xZOne** [comment]
> I'd like to understand why this is necessary. We don't use FML for managing all threads in the process. So the use of the TLS slot becomes problematic. Can you add a link to the issue you are trying to fix?

Thank you for your reply. Actually, there is no corresponding issue on Github. Thread-related information is required for debugging in our custom Flutter engine. You can close it anytime. :)

Could you elaborate on why the TLS slot is problematic here? Thanks.

**Comment by 0xZOne** [comment]
> > I'd like to understand why this is necessary. We don't use FML for managing all threads in the process. So the use of the TLS slot becomes problematic. Can you add a link to the issue you are trying to fix?
> 
> Thank you for your reply. Actually, there is no corresponding issue on Github. Thread-related information is required for debugging in our custom Flutter engine. You can close it anytime. :)
> 
> Could you elaborate on why the TLS slot is problematic here? Thanks.

In fact, there are also scenarios where the thread name is obtained in Flutter.

**Comment by chinmaygarde** [comment]
Since there are no users of this API in the OSS engine, I am closing this PR. We can reopen this if we need this functionality.

**Pull request closed without merging**